src/components/settings/SettingsPage.js

```diff
@@ -58,6 +64,13 @@ const SettingsPage = ({ desktop }) => {
   const [overrideDataLanguage, setOverrideDataLanguage] = useState(false)
   const { t, i18n } = useTranslation()
 
+  const updateUnitsSetting = (index) => {
+    for (const [key, value] of Object.entries(METRICS_OPTIONS)) {
+      if (index === value) {
+        dispatch(updateSettings({ ['distanceUnit']: key }))
+      }
+    }
```

Comment:
So you could avoid having this for loop, since onChange would return value directly.